system/ee/legacy/libraries/Template.php

```diff
@@ -232,15 +232,40 @@ public function fetch_and_parse($template_group = '', $template = '', $is_embed
         //  - Modify template after tag parsing
         //
         if (ee()->extensions->active_hook('template_post_parse') === true) {
-            $this->final_template = ee()->extensions->call(
-                'template_post_parse',
-                $this->final_template,
-                ($is_embed || $is_layout), // $is_partial
-                $site_id
-            );
+            // Populate the $currentTemplateInfo array
+            $currentTemplateInfo = array();
+            if ($this->templates_loaded) // don't do this if we don't have any template info!
+            {
+                $currentTemplateInfo = array(
+                    'template_name' => $template,
+                    'template_group' => $template_group,
+                );
+            }
+            // Add a conditional to adjust what is returned by function depending on what kind of template we are dealing with
+            if ($this->final_template)
```

Comment:
I think this might work wrong if `$this->final_template` is an empty string, no?